### ISSUE TITLE
I am using5600xt，when I use "hevc_amf" in ffmpeg, its always broken，how can i use hardware encoder in ffmpeg

### ISSUE THREAD

**Issue by angussleeping** [GPUOpen-LibrariesAndSDKs/AMF#273]

**Comment by rhutsAMD** [comment]
Which OS are you using? HEVC is supported in this case on Windows but not on Linux.

On Windows 10, I was able to use the FFmpeg hevc_amf encoder with the following command line options on an RX5600 XT to encode a sample YUV video where the output looked reasonable.
`ffmpeg.exe -f rawvideo -pix_fmt yuv420p -s:v 1920x1080 -r 25 -i big_buck_bunny_1920x1080.yuv -c:v hevc_amf output_big_buck_ffmpeg_amf_1920x1080.hevc`

You can identify which encoder is used in the stream mapping that is displayed in the console output:

```
Stream mapping:
  Stream #0:0 -> #0:0 (rawvideo (native) -> hevc (hevc_amf))
```
	
To view the other available options such as the usage preset which I kept as default, you can run this command:
`ffmpeg.exe -h encoder=hevc_amf`

What are the command line options and what is the FFmpeg version that you are using?

**Comment by angussleeping** [comment]
I am using win10 and ffmpeg 4.4.1, and I'm using axiom(1.9.6.1-alpha) and vs video convertor(7.16.3) to encoder my vedio, I'm not professional but a fun, I cannot remember the command exactly，but I remember that it always broken when encoding the video stream，in axiom I change the “container” to “MKV”，”video codec“ to “HEVC AMF”，“HW accel” to “on”，“decode” to “D3D11VA”，“transcode” to “AMD AMF”，”Audoi“ and "Subtitles" to "none" to get rid of the influence of other factors, but the graphics card is still break down.

**Comment by rhutsAMD** [comment]
Using the same FFmpeg version, axiom version, Win10, RX 5600 XT with the latest 21.11.3 driver and the settings you have provided, the issue does not reproduce. Could you please provide the exact command line that you are using as well as the input file(s)? Axiom will generate the full command line if you press the "Script" button.

**Comment by angussleeping** [comment]
sorry, I was busy last week
I has delate that file already, and just now I try another video with Axiom, but it dosenot broken, I don't know why.
the command is:
"C:\Users\Administrator\Desktop\Axiom.FFmpeg\ffmpeg\bin\ffmpeg.exe" 

-y 

-hwaccel d3d11va 

-i "F:\余罪.第一季.Yu.Zui.2016.S01.EP12.WEB-DL.1080p.H264.AAC2.0-DHTCLUB.mp4" 

-c:v hevc_amf 
-quality balanced -b:v 2527K 
-pix_fmt yuv420p 
-vf "scale=trunc(iw/2)*2:trunc(ih/2)*2" 
-pass 1 

-an 

-sn 

-f matroska 

-threads 12 

NUL 

&& 

"C:\Users\Administrator\Desktop\Axiom.FFmpeg\ffmpeg\bin\ffmpeg.exe" 

-y 

-hwaccel d3d11va 

-i "F:\余罪.第一季.Yu.Zui.2016.S01.EP12.WEB-DL.1080p.H264.AAC2.0-DHTCLUB.mp4" 

-c:v hevc_amf 
-quality balanced -b:v 2527K 
-pix_fmt yuv420p 
-vf "scale=trunc(iw/2)*2:trunc(ih/2)*2" 
-map 0:v? 
-pass 2 

-c:a copy 
-map 0:a? 
-sn 

-map_chapters 0 

-map_metadata 0 

-f matroska 

-threads 12 

"F:\余罪.第一季.Yu.Zui.2016.S01.EP12.WEB-DL.1080p.H264.AAC2.0-DHTCLUB.mkv"

this time it dosenot broken, but the command is the same,i will try another video again, please wait for 1 or 2 days,thanks

**Comment by angussleeping** [comment]
I tried again, it broken again, ffmpeg shows that 
"[swscaler @ 0000020f7092e4c0] deprecated pixel format used, make sure you did set range correctly
[AVHWFramesContext @ 0000020f70b18580] Could not create the staging texture (887a0005)
[h264 @ 0000020f705a2700] Failed to transfer data to output frame: -1313558101.
Error while processing the decoded data for stream #0:0"
The same time Radeon Software broken, ffmpeg command is 
""C:\Users\Administrator\Desktop\Axiom.FFmpeg\ffmpeg\bin\ffmpeg.exe" 
-y 
-hwaccel d3d11va 
-i "F:\Armour of god II - Blu-ray™.mkv" 
-c:v hevc_amf 
-quality balanced -b:v 25546K 
-pix_fmt yuv420p 
-vf "scale=trunc(iw/2)*2:trunc(ih/2)*2" 
-pass 1 
-an 
-sn 
-f matroska 
-threads 12 
NUL 
&& 
"C:\Users\Administrator\Desktop\Axiom.FFmpeg\ffmpeg\bin\ffmpeg.exe" 
-y 
-hwaccel d3d11va 
-i "F:\Armour of god II - Blu-ray™.mkv" 
-c:v hevc_amf 
-quality balanced -b:v 25546K 
-pix_fmt yuv420p 
-vf "scale=trunc(iw/2)*2:trunc(ih/2)*2" 
-map 0:v? 
-pass 2 
-c:a copy 
-map 0:a? 
-c:s copy 
-map 0:s? 
-map_chapters 0 
-map_metadata 0 
-f matroska 
-threads 12 
"F:\Armour of god II - Blu-ray™ (1).mkv""
and the bug report is below
[AMD Bug Reporting Form Submitted Data 20211130T123205.zip](https://github.com/GPUOpen-LibrariesAndSDKs/AMF/files/7622952/AMD.Bug.Reporting.Form.Submitted.Data.20211130T123205.zip)
please help me to solve this problem, thank you!!!

**Comment by rhutsAMD** [comment]
887a0005 is the error code for DXGI_ERROR_DEVICE_REMOVED which suggests a crash/TDR. Please try to change the pixel format to NV12 `-pix_fmt nv12`. I tried your cmd line options with the “Big_Buck_Bunny_1080_10s_30MB.mkv” sample video using both nv12 and yuv420p pixel format options on an RX 5600 XT with the latest 21.12.1 driver, but still could not reproduce the issue.

**Comment by angussleeping** [comment]
I update the driver to 21.12.1, then I tried again, this time I try to change pixel formate to nv12, it runs normal with a video that format configuration(profile) is main@L3.2
I tried another video that Format configuration(profile) is main10@L5.1@high, it shows that 
"[AVHWFramesContext @ 0000024b00849f80] Could not create the staging texture (887a0005)
[hevc @ 0000024b00412f00] Failed to transfer data to output frame: -1313558101.
Error while processing the decoded data for stream #0:0"
then I try to change pixel format to all the 5 options, it always shows the same error and radeon software broken.
Could it be a format configuration problem?
